### ISSUE TITLE
Adhere to WCAG2.1 accessibility guidelines

### ISSUE THREAD

**Issue by samccann** [ansible-community/community-website#9]
Our site should be accessible to all users. 

- [x] Consider accessibility during design and implementation of the wireframe
- [x] Find open accessibility testing tools we can apply to the new website
- [x] Complete [WCAG assessment for the homepage](https://github.com/ansible-community/community-website/issues/9#issuecomment-1746972479)
- [x] Complete [WCAG assessment for blog posts](https://github.com/ansible-community/community-website/issues/9#issuecomment-1750339110)

**Comment by oraNod** [comment]
> This resource will help us set guidelines https://www.ibm.com/able/

Thanks @samccann There is also https://ux.redhat.com/ available.

We've reached a point where the WIP version of the web site matches the wireframe fairly closely and the structure is in place with all content loaded and css classes defined. It's time for an update on this issue and to discuss some next steps.

Firstly I feel that we can mark the "Consider accessibility at all steps of the process" complete. Actually I'm going to change the wording on that one otherwise we'll never be able to close it. There are still more steps in this process to come and accessibility is an ongoing commitment.

As far as the WIP version goes we still need to add all the media breakpoints for mobile responsiveness but we have used a 12 column grid layout and have kept the structure of the site in logical order. The content is mostly text and should be easily navigable for users of assistive technology like screen readers etc. The layout of each section follows a logical order (heading 3 follows heading 2) and there is no use of semantic markup for layout. Any images are clearly labelled with meaningful alternative descriptions. In no case is an image or other visual object the sole means of communication to the viewer.

There are probably some elements that are missing labels but I can use mozilla tools to navigate through the site and detect any violations. We'll definitely need to make sure we can't detect any violations before the launch. 

Looking at this: https://www.w3.org/TR/WCAG21/#dfn-accessibility-supported

We can certainly claim 2.1 -- supported natively in widely-distributed user agents that are also accessibility supported (such as HTML and CSS)

For the actual accessibility testing is this something we can ask Red Hat to facilitate? I think that is the thing we need to be fully conformant with WCAG2.1. But if we do our own testing is that robust enough?

Here's a screengrab of what I see with the mozilla accessibility console:

![image](https://github.com/ansible-community/community-website/assets/32749632/af9ee675-432d-4fb8-b21c-e9e90111b7a4)

**Comment by oraNod** [comment]
@samccann I've updated the issue description with more complete details about the work ahead. Can you please take a look at that as well? Cheers.

**Comment by oraNod** [comment]
Tools we can use for accessibility testing (other than mozilla dev tools):

- https://wave.webaim.org/
- https://github.com/pa11y/pa11y
- https://github.com/marketplace/axe-linter
- https://github.com/dequelabs/axe-core
- https://github.com/nickcolley/jest-axe

It's also worth looking through https://webaim.org/

**Comment by oraNod** [comment]
WCAG assessment for the homepage

- [ ] Check for WCAG2.1 conformity so that [information and user interface components are presentable to users in ways they can perceive](https://www.w3.org/TR/WCAG21/#perceivable)
- [ ] Check for WCAG2.1 conformity so that [user interface components and navigation are operable](https://www.w3.org/TR/WCAG21/#operable)
- [ ] Check for WCAG2.1 conformity so that [information and the operation of user interface are understandable](https://www.w3.org/TR/WCAG21/#understandable)
- [ ] Check for WCAG2.1 conformity so that [site content is robust](https://www.w3.org/TR/WCAG21/#robust)
- [ ] Test site navigation using appropriate assistive technology (AT).

**Comment by oraNod** [comment]
I have completed accessibility testing of the homepage using the WCAG2.1 guidelines and checklists. To the best of my ability I have tested navigability and robustness of the site using both my keyboard and text to speech functionality on an iPad.

I have verified that all colour contrasts pass WCAG A and WCAG AA tests with the following tools:

- https://webaim.org/resources/contrastchecker/
- https://webaim.org/resources/linkcontrastchecker/

I used the following evaluation tool to ensure that the site does not trigger any alerts or errors related to WCAG standards: https://wave.webaim.org/report#/https://ansible-community-website.readthedocs.io/en/latest/

I used `axe-core` tooling for additional quality checks and to adhere to best practices:

VSCode extension (linter): https://marketplace.visualstudio.com/items?itemName=deque-systems.vscode-axe-linter
Browser extension: https://www.deque.com/axe/browser-extensions/
 
Note that https://github.com/dequelabs/axe-core provides `ci` tools to automate testing. This tooling is more geared towards NodeJS and web applications than static sites. Now that we have a fully accessible baseline we can just do ad-hoc testing and look for issues in pull requests.

To ensure that continues to happen I'll update the main README with instructions on how to test for accessibility.

**Comment by oraNod** [comment]
### WCAG Assessment checklist for blog posts

- [ ] Check for WCAG2.1 conformity so that [information and user interface components are presentable to users in ways they can perceive](https://www.w3.org/TR/WCAG21/#perceivable), for example:
-- [ ] Ensure that all images have meaningful and descriptive alternative text.
-- [ ] Ensure that no images are the sole means to convey information.
-- [ ] Check for contrast violations and remediate.
-- [ ] Check for missing labels or alternative text and remediate.
- [ ] Check for WCAG2.1 conformity so that [user interface components and navigation are operable](https://www.w3.org/TR/WCAG21/#operable), for example:
-- [ ] Disable or ignore icons or other decorative elements for assistive technology devices.
-- [ ] Check that section headings are used to organize content appropriately.
- [ ] Check for WCAG2.1 conformity so that [information and the operation of user interface are understandable](https://www.w3.org/TR/WCAG21/#understandable), for example:
-- [ ] Ensure that links open to the expected target.
-- [ ] Check buttons are predictable and behave as expected.
-- [ ] Ensure that any user input is clearly labelled and has instructions.
- [ ] Check for WCAG2.1 conformity so that [site content is robust](https://www.w3.org/TR/WCAG21/#robust), for example:
-- [ ] Elements have complete start and end tags
-- [ ] Elements are nested according to their specifications
-- [ ] Elements do not contain duplicate attributes
-- [ ] Any IDs are unique
- [ ] Test site navigation using appropriate assistive technology (AT).
